app/components/message_component.html.erb

```diff
@@ -36,7 +36,11 @@
     </div>
     <% if @message.visualizable_body? %>
       <div class="flex flex-col justify-stretch items-start relative gap-4 p-6 border-t-0 border-r-0 border-b border-l-0 border-gray-200">
-      <% if @message.html_visualization %>
+      <% if @message.metadata&.dig('delivery_notification')&.dig('consignment')&.dig('type') == 'Doc.GeneralAgendaReport' %>
+            <%= render DeliveryNotificationMessageBodyComponent.new(message: @message) %>
+      <% elsif @message.metadata&.dig('delivery_notification')&.dig('consignment')&.dig('type') == 'Doc.GeneralAgendaFiction' %>
+            <%= render DeliveryNotificationFictionMessageBodyComponent.new(message: @message) %>
```

Comment:
Toto je vec z upvs domeny cize by som jej dal prefix `Upvs::` zaroven toto vies asi ulozit do tej spravy ako nejaky message_type (mozno dokonca vieme rovno ukladat pri parsovani?)

Reply:
Dalo by sa ukladat, ale oplati sa to, ked to tam tak ci tak mame (iba hlbsie)? @jsuchal

Reply:
Nemusime teraz riesit ale message_component by nemal vediet nic o UPVS domene, preto dava zmysel, aby sa tu pouzila nejaka polymorfia namiesto if. Vyhladovo by som to potom upratal